app/Services/Registrar.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Gladiator\Services;
+
+use Gladiator\Models\User;
+use Gladiator\Services\Northstar\Northstar;
+use Gladiator\Services\Northstar\Exceptions\NorthstarUserNotFoundException;
+
+class Registrar
+{
+    /**
+     * Northstar instance.
+     *
+     * @var \Gladiator\Services\Northstar\Northstar
+     */
+    protected $northstar;
+
+    /**
+     * Create new Registrar instance.
+     *
+     * @param Northstar $northstar
+     */
+    public function __construct(Northstar $northstar)
+    {
+        $this->northstar = $northstar;
+    }
+
+    /**
+     * Resolve user within system or create new user.
+     *
+     * @param  array  $credentials
+     * @return \Gladiator\Models\User
+     */
+    public function findOrCreate($credentials)
+    {
+        $account = $this->hasAccountInSystem($credentials['type'], $credentials['key']);
+
+        if ($account instanceof User) {
+            $user = $account;
+
+            session()->flash('status', 'User already exists!');
```

Comment:
Ideally the service class shouldn't have any presentation stuff in it (since this flash messaging will only work on web requests). It might be a good idea to break this into two separate methods like `findExistingUser` and `createUser`, so you can present the results of each step based on context (e.g. switching status code `200` to `201` for the API).